src/main.cpp

```diff
@@ -6792,6 +6793,107 @@ ThresholdState VersionBitsTipState(const Consensus::Params& params, Consensus::D
     return VersionBitsState(chainActive.Tip(), params, pos, versionbitscache);
 }
 
+bool LoadMempool(void)
+{
+    FILE* filestr = fopen((GetDataDir() / "mempool.dat").c_str(), "r");
+    CAutoFile file(filestr, SER_DISK, CLIENT_VERSION);
+    if (file.IsNull()) {
+        return error("Failed to open mempool file from disk");
+    }
+
+    int64_t count = 0;
+
+    try {
+        uint64_t version;
+        file >> version;
+        if (version != 1) {
+            return false;
+        }
+        uint64_t num;
+        file >> num;
+        double prioritydummy = 0;
+        while (num--) {
+            CTransaction tx;
+            int64_t nTime;
+            int64_t nFeeDelta;
+            file >> tx;
+            file >> nTime;
+            file >> nFeeDelta;
```

Comment:
That would be ugly. It's an internal data structure with various indexes associated to it. It's not the mempool entry that is stored - if it were, we wouldn't need ATMP to load it back.